src/docbuilder/rustwide_builder.rs

```diff
@@ -446,6 +446,15 @@ impl RustwideBuilder {
                         Err(err) => warn!("{:#?}", err),
                     }
 
+                    // delete eventually existing files from pre-archive storage.
+                    // we're doing this in the end so eventual problems in the build
+                    // won't lead to non-existing docs.
+                    for prefix in &["rustdoc", "sources"] {
+                        let prefix = format!("{}/{}/{}/", prefix, name, version);
+                        log::debug!("cleaning old storage folder {}", prefix);
+                        self.storage.delete_prefix(&prefix)?;
```

Comment:
Can you only delete these if the build was successful? Otherwise if the new build fails we'll end up with *neither* old nor new docs.

Reply:
yep, I'll change it. 

My initial thought was that we also overwrite the build-result on the release when the build fails, 
but it seems like that the build-status actually doesn't affect how docs are rendered, these only have to exist.

Reply:
@jyn514 what would be an easy way to force a failed build to I can write a test for this behavior?

Reply:
Build a binary crate, maybe? We don't generate docs for those currently.

Otherwise you could build a local crate that you know doesn't compile (or find one on crates.io).